consul/leader.go

```diff
@@ -76,6 +76,19 @@ func (s *Server) leaderLoop(stopCh chan struct{}) {
 	var reconcileCh chan serf.Member
 	establishedLeader := false
 
+	reassert := func() error {
```

Comment:
I thought this ended up cleaner than having an error write to the channel and a `continue` on each line of the function - made the `select` simpler.